### PR TITLE
Added the code for saving TODOs locally

### PR DESCRIPTION
**Pull request by raj5036**
Please review the changes.

### CODE REVIEW

**Comment by EdgeKing810** [comment]
Hey @raj5036. I didn't test yet but from I can see from the diff, this task is not complete. Here's a couple of points to work on:

* In `App.js`, create a constant variable in the `addTodoListener` function, which you'll give the value `v1().toString()`. Substitute that variable in the value of the `key` field where the new Todo is being saved to state.
* Move the function for saving todos to `App.js`.
* Reference the function call to that ^ function in the `addTodoListener` function, passing in the variable holding the key as second parameter instead.
* If the data is being successfully being saved correctly when a Todo is created, it needs to also be fetched from the local storage when the app is opened. Implement this in `App.js` itself.

**Comment by raj5036** [comment]
Okay.

**Comment by raj5036** [comment]
Changed the code as per the requirements. Please review the code.

**Comment by raj5036** [comment]
Changed the code as per the requirements. Please review the changes .

**Comment by EdgeKing810** [comment]
@raj5036 Unfortunately this is still incorrect. Why did you remove the function for saving todos to the localStorage? I only asked you to move it into App.js and use the same key that was being generated using the `v1` module instead of the incremented state variable you were using. Please read all the bullet points in my previous comment.

**Comment by raj5036** [comment]
Okay

**Comment by raj5036** [comment]
Added those changes. Is there anything I am still missing ?

**Comment by EdgeKing810** [comment]
Good to see the improvements! Last couple of things:

When you call the `storeLocally` function, save the `todos` array as a whole instead with a hardcoded id or something. Maybe consider using something like `JSON.Stringify` to wrap the whole thing as a string and store it by a predefined key such as `"localTodos"`.

Move the contents of the `showTodos` function in a `useEffect` that will run only once, and it should now fetch the string from the `AsyncStorage` that corresponds to the predefined key (`"localTodos"`). This is done because we only want it to fetch stuff from the local storage when the apps starts up. Then, you should `JSON.Parse` the returned string and pass it to `setTodos`. 

Don't hesitate to ask questions if I wasn't clear enough. I'll respond in less than 24 hours.

**Comment by raj5036** [comment]
Is it Okay?

**Comment by raj5036** [comment]
And will this PR be counted towards my Hacktoberfest contribution or should I create a new PR? @EdgeKing810

**Comment by EdgeKing810** [comment]
Yes, it will be counted. One final change and you're good to go @raj5036!

Remove this block:
```
parsedValues.forEach(parsedValue=>{
    console.log(`${parsedValue.key} : ${parsedValue.name}`);
});
```

and instead, just update the state directly; `setTodos(parsedValues);`

**Comment by raj5036** [comment]
Okay

**Comment by raj5036** [comment]
Okay.

**Comment by raj5036** [comment]
Okay check it.

**Comment by raj5036** [comment]
Is it Okay? @EdgeKing810

File: App.js [hunk 1]
```diff
@@ -25,9 +26,38 @@ export default function App() {
...
+            const value=await AsyncStorage.getItem('localTodos');
+            if(value!=null){
+                let parsedValues=JSON.parse(value);
+                setTodos(parsedValues);
+            }
+        }
+        catch(error){
+            console.log(error);
+        }
+    },[todos]);
```
Comment (EdgeKing810) [inline App.js]: Remove `todos` from the array, and put a blank array instead.

**Comment by EdgeKing810** [comment]
Check my review. It should be `[]` instead of `[todos]`. This is because it should fetch todos from the local storage only when the app starts up and not each time a change to the todos has been done.

File: App.js (hunk 1 above)
Comment (raj5036) [reply App.js]: Okay

**Comment by raj5036** [comment]
Did it. @EdgeKing810

**Comment by EdgeKing810** [comment]
LGTM! Thanks @raj5036 :tada:

**Pull request merged**